From 16050b74eaf55cc7b4b06f29ebcbf3a76ebd1d26 Mon Sep 17 00:00:00 2001
From: Nils DEYBACH <68770774+ndeybach@users.noreply.github.com>
Date: Fri, 9 Jan 2026 17:11:59 +0100
Subject: [PATCH] fix: remove libm when under WIN32

Do not add the libm entry under mindows or macos as it would not be present and errors out.

Inspired by 3d0a160d34 - Remove libm dependency when building with MSVC (#4314) by  ayeteadoe <ayeteadoe@gmail.com>.
---
 lib/jxl.cmake | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/lib/jxl.cmake b/lib/jxl.cmake
index 59b2cca4..0c78059c 100644
--- a/lib/jxl.cmake
+++ b/lib/jxl.cmake
@@ -269,12 +269,17 @@ install(TARGETS jxl
 set(JPEGXL_LIBRARY_REQUIRES
     "libhwy libbrotlienc libbrotlidec libjxl_cms")
 
+# MSVCRT bundles math functions so no explicit libm dependency is required
 if (BUILD_SHARED_LIBS)
   set(JPEGXL_REQUIRES_TYPE "Requires.private")
-  set(JPEGXL_PRIVATE_LIBS "-lm ${PKGCONFIG_CXX_LIB}")
+  if(NOT WIN32 AND NOT APPLE)
+    set(JPEGXL_PRIVATE_LIBS "-lm ${PKGCONFIG_CXX_LIB}")
+  endif()
 else()
   set(JPEGXL_REQUIRES_TYPE "Requires")
-  set(JPEGXL_PUBLIC_LIBS "-lm ${PKGCONFIG_CXX_LIB}")
+  if(NOT WIN32 AND NOT APPLE)
+    set(JPEGXL_PUBLIC_LIBS "-lm ${PKGCONFIG_CXX_LIB}")
+  endif()
 endif()
 
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/jxl/libjxl.pc.in"
-- 
2.52.0